TDL-09-shell-EC2/deploy_instance.sh

```diff
@@ -0,0 +1,229 @@
+declare -a LIST_OSTYPE
+LIST_OSTYPE=('amzn' 'amzn2' 'centos6' 'centos7' 'ubuntu')
+declare -a LIST_INSTANCE_TYPE
+LIST_INSTANCE_TYPE=('t2.micro' 't2.small' 't3.micro')
+instance_type='t2.micro' #default value
+
+#######################################
+# Print a message in a given color.
+# Arguments:
+#   Color. eg: green, red
+#######################################
+function print_color(){
+  NC='\033[0m' # No Color
+  case $1 in
+    "green") COLOR='\033[0;32m' ;;
+    "red") COLOR='\033[0;31m' ;;
+    "*") COLOR='\033[0m' ;;
+  esac
+
+  echo -e "${COLOR} $2 ${NC}"
+}
+
+#######################################
+# Check item in a list.
+# Arguments:
+#   list item
+#######################################
+function list_include_item {
+  list="$1"
+  item="$2"
+  if [[ $list =~ (^|[[:space:]])"$item"($|[[:space:]]) ]] ; then
+    result=0  # yes, list include item
+  else
+    result=1
+  fi
+  echo $result
+  return $result
+}
+
+#######################################
+# Show how to use the deploy_instance.sh.
+# Arguments:
+#   usage_type. eg: all, ostype, agedtime, instance_type.
+#######################################
+function usage(){
+  case "$1" in
+    all)
+      print_color "red" "To deploy your instance, please run deploy_instance.sh as follows:"
+      print_color "red" "./deploy_instance.sh ostype agedtime [instance_type]"
+      print_color "red" "The value range of the arguments: "
+      print_color "red" "1. valid ostype: amzn, amzn2, centos6, centos7, ubuntu"
+      print_color "red" "2. agedtime: 1-7 "
+      print_color "red" "3. instance_type: default t2_micro, valid instance_type: t2.micro, t2.small, t3.micro"
+      ;;
+    ostype)
+      print_color "red" "Please input valid ostype: amzn, amzn2, centos6, centos7, ubuntu"
+      ;;
+    agedtime)
+      print_color "red" "Please input valid agedtime: 1-7 "
+      ;;
+    instance_type)
+      print_color "red" "Please input valid instance_type: default t2_micro, other valid instance_type: t2.micro, t2.small, t3.micro"
+      ;;
+    *)
+      error "Unexpected expression '$1'"
+      ;;
+  esac
+}
+
+# check input
+print_color "green" "-----------------------------Your input:-------------------------------"
+echo "ostype=$1,agetime=$2,instance_type=$3"
+print_color "green" "-----------------------------Check input-------------------------------"
+
+# if less than 2 parameters from command line, print usage() and exit
+if [[ $# -lt 2 || $# -gt 3 ]];  then
+  usage 'all'
+  exit
+fi
+
+# check whehter ostype is a valid option, if ostype is an invalid ostype, print usage() and exit 
+list_include_item "${LIST_OSTYPE[*]}" "$1"
+result=$?
+if (( result == 1 )); then
+  usage 'ostype'
+  exit
+fi
+ostype=$1
+
+# check whether 1 <= agedtime <=7, otherwise iprint usage() and exit
+if (( $2 <1 || $2 > 7)); then
+  usage 'agedtime'
+  exit
+fi
+agedtime=$2
+
+# check whether instance_type is valid, otherwise iprint usage() and exit
+if [[ $# -eq 3 ]];  then
```

Comment:
instance_type=${3:-t2.micro} ?

which will set up the default value for instance_type

Reply:
Good idea. Thanks